Score.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+#include "Score.h"
+#include <fstream>
+#include <iostream>
+#include <string>
+using namespace std;
+
+
+   //constructeur
+    Score::Score(int score) : m_score(score)
+{
+
+}
+    
+void Score::setScore(int s){
+    int* ptr = &m_score;
+    *ptr =  m_score + s;//just to try pointers
+}
+
+    void Score::display() const {
+//        cout<<"\x1B[2J\x1B[0;0f"; // go down in terminal
+        cout<<m_score<<endl;
+    }
+    
+     int Score::getFileScore(){//A CONTINUER POUR EDITER LE FICHIER TXT
+	fstream my_file;
+	my_file.open("scores.txt",ios::in);
+        int highscore;
+        my_file>>highscore;
+        return highscore; 
+     }
+     
+     
+    void Score::editScore(){//A CONTINUER POUR EDITER LE FICHIER TXT
+	fstream my_file;
+	my_file.open("scores.txt",ios::out);
+        int highscore = getFileScore();
+//        my_file>> highscore; 
+	if (!my_file) {
+		cout << "File not created!";

```

Comment:
throw an exception

Reply:
what do you mean by exception ?

Reply:
I can provide you with two useful links: 

- [w3school tutorial](https://www.w3schools.com/cpp/cpp_exceptions.asp)
- [isocpp](https://isocpp.org/wiki/faq/exceptions)